### ISSUE TITLE
Invalid compiled template in 3.1.32-dev

### ISSUE THREAD

**Issue by robo2bobo** [smarty-php/smarty#429]
I am still using 3.1.27, because every time I try one of the newer versions something is broken. This time, I switched to 3.1.32-dev latest SVN, to see if things would work better, but I immediately got a Smarty error.

My site's index.php loads a parent template.html and a child template homepage.html, like this:
```php
$smarty->display('extends:template.html|homepage.html', $_action);
```

The parent template.html has an empty default block:
```
{block name=content}{/block}
```

The child homepage.html has a select tag block:
```
{block name=content}
<select class="form-control" name="country">
	{include 'select-country.html' caching}
</select>
{/block}
```

The included select-country.html has only option tags:
```
<option value=""></option>
<option value="Austria">Austria</option>
```

The above works fine in 3.1.27, but when I try 3.1.32-dev without any changes, I get the following error:
```
Fatal error: Uncaught --> Smarty: Invalid compiled template for 'file:select-country.html' <-- thrown in /public_html/smarty/libs/sysplugins/smarty_template_resource_base.php on line 123 
```

I am not sure what is wrong, maybe something changed in the way blocks are inherited? I would appreciate some help please.

Thank you.

**Comment by uwetews** [comment]
I was not able to reproduce your problem.
Try again after manually deleting all cached and compiled template files

**Comment by robo2bobo** [comment]
Here it is always reproducible. Maybe something in the way I call Smarty is affecting this bug, like the setTemplateDir() which is an array for several paths? or maybe my custom Smarty_Security()? I am not sure.

I managed to avoid this bug by accident, I changed the parameter "caching" to "inline" and now the template works fine without a problem. I am not sure why that is, but at least it allows me to make further tests with v3.1.32-dev.
